agents/smurf_file_emulator/smurf_file_emulator.py

```diff
@@ -92,6 +220,71 @@ def get_data_frame(self, start, stop):
         return fr
 
 
+class DataStreamer:
+    """
+    Helper class for streaming G3 data
+    """
+
+    def __init__(self, stream_id, sample_rate, tune, timestreamdir,
+                 file_duration, frame_len):
+        self.frame_gen = G3FrameGenerator(stream_id, sample_rate, tune)
+        self.session_id = self.frame_gen.session_id
+        self.stream_id = stream_id
+        self.timestreamdir = timestreamdir
+        self.seq = 0
+        self.file_duration = file_duration
+        self.file_start = 0
+        self.writer = None
+        self.file_list = []
+        self.frame_len = frame_len
+
+    def get_g3_filename(self):
+        """
+        Returns the file path for a g3-file with specified session id and seq
+        idx.
+        """
+        timecode = f"{self.session_id}"[:5]
+        subdir = os.path.join(self.timestreamdir, timecode, self.stream_id)
+        filepath = os.path.join(subdir, f"{self.session_id}_{self.seq:0>3}.g3")
+        return filepath
+
+    def new_file(self):
+        self.end_file()
+        fname = self.get_g3_filename()
+        os.makedirs(os.path.dirname(fname), exist_ok=True)
+        self.writer = core.G3Writer(fname)
+        if self.seq == 0:
+            self.writer(self.frame_gen.get_obs_frame())
+            self.writer(self.frame_gen.get_status_frame())
+        self.file_start = time.time()
+        self.file_list.append(fname)
+        self.seq += 1
+
+    def end_file(self):
+        if self.writer is not None:
+            self.writer(core.G3Frame(core.G3FrameType.EndProcessing))
+
+    def write_next(self, wait=True):
+        start = time.time()
+        if (start - self.file_start > self.file_duration) or (self.writer is None):
+            self.new_file()
+        time.sleep(self.frame_len)
+        stop = time.time()
+        self.writer(self.frame_gen.get_data_frame(start, stop))
+
+    def stream_between(self, start, stop, wait=False):
+        frame_starts = np.arange(start, stop, self.frame_len)
+        frame_stops = frame_starts + self.frame_len
+        self.new_file()
+        for t0, t1 in zip(frame_starts, frame_stops):
+            if wait:
+                now = time.time()
+                if now < t1:
+                    time.sleep(t1 - now)
+            self.writer(self.frame_gen.get_data_frame(start, stop))
+        self.end_file()
```

Comment:
Can you add docstrings for these three methods (`stream_between`, `write_next`, and `end_file`)?